pkg/controller/clusterconfig/clusterconfig_controller.go

```diff
@@ -91,28 +91,28 @@ func (r *ReconcileClusterConfig) Reconcile(request reconcile.Request) (reconcile
 	// Validate the cluster config
 	if err := network.ValidateClusterConfig(clusterConfig.Spec); err != nil {
 		log.Printf("Failed to validate Network.Spec: %v", err)
-		r.status.SetDegraded(statusmanager.ClusterConfig, "InvalidClusterConfig",
+		r.status.SetDegraded(statusmanager.ClusterConfig, true, "InvalidClusterConfig",
 			fmt.Sprintf("The cluster configuration is invalid (%v). Use 'oc edit network.config.openshift.io cluster' to fix.", err))
 		return reconcile.Result{}, err
 	}
 
 	operatorConfig, err := r.UpdateOperatorConfig(context.TODO(), *clusterConfig)
 	if err != nil {
 		log.Printf("Failed to generate NetworkConfig CRD: %v", err)
-		r.status.SetDegraded(statusmanager.ClusterConfig, "UpdateOperatorConfig",
+		r.status.SetDegraded(statusmanager.ClusterConfig, true, "UpdateOperatorConfig",
 			fmt.Sprintf("Internal error while converting cluster configuration: %v", err))
 		return reconcile.Result{}, err
 	}
 
 	if operatorConfig != nil {
 		if err := apply.ApplyObject(context.TODO(), r.client, operatorConfig); err != nil {
 			log.Printf("Could not apply operator config: %v", err)
-			r.status.SetDegraded(statusmanager.ClusterConfig, "ApplyOperatorConfig",
+			r.status.SetDegraded(statusmanager.ClusterConfig, true, "ApplyOperatorConfig",
 				fmt.Sprintf("Error while trying to update operator configuration: %v", err))
 			return reconcile.Result{}, err
 		}
 	}
 
-	r.status.SetNotDegraded(statusmanager.ClusterConfig)
+	r.status.SetDegraded(statusmanager.ClusterConfig, false, "", "")
```

Comment:
I don't like having the unnecessary `"", ""` there. That's why I had separate `SetDegraded` and `SetNotDegraded` to begin with.

You can keep the internal `setDegraded` with a bool but still have separate public `SetDegraded` and `SetNotDegraded` methods